app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -64,4 +107,48 @@ class HomeFragmentPresenter @Inject constructor(
       }
     })
   }
+
+  private val topicListSummaryResultLiveData: LiveData<AsyncResult<TopicList>> by lazy {
+    topicListController.getTopicList()
+  }
+
+  private fun subscribeToTopicList() {
+    getAssumedSuccessfulTopicList().observe(fragment, Observer<TopicList> { result ->
+      if (getPromotedStoryViewModel() != null) {
+        getPromotedStoryViewModel()!!.setPromotedStory(result.promotedStory)
+        itemList.add(getPromotedStoryViewModel()!!)
+      }
+      itemList.addAll(result.topicSummaryList)
+      topicListAdapter.notifyDataSetChanged()
+    })
+  }
+
+  private fun getAssumedSuccessfulTopicList(): LiveData<TopicList> {
+    // If there's an error loading the data, assume the default.
+    return Transformations.map(topicListSummaryResultLiveData) { it.getOrDefault(TopicList.getDefaultInstance()) }
+  }
+
+  private fun subscribeToUserAppHistory() {
+    getUserAppHistory()!!.observe(fragment, Observer<UserAppHistory> { result ->
+      getUserAppHistoryViewModel().setAlreadyAppOpened(result.alreadyOpenedApp)
+      itemList.add(0, getUserAppHistoryViewModel())
+      topicListAdapter.notifyDataSetChanged()
+    })
+  }
+
+  private fun getUserAppHistory(): LiveData<UserAppHistory>? {
```

Comment:
Why the nullable return value? Doesn't Transformations.map() guarantee a non-null return if the LiveData passed in is non-null?

Reply:
Removed nullable.